### DISCUSSION TITLE
Automiss for near hits

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#26777] [Ideas]
https://youtu.be/QclB1bnGkVU

I recently had this happen to me and I figure there should be something in place to prevent my single mistake from escalating into multiple hits to my score.

What happened?
I missed the correct circle by pixels, causing me to hit the circle under it well before it was time to do so, causing me to miss both at once. This also would have caused me to miss any more notes within a burst/stream as when I would go to click the circle I just early tapped, I would delete my next note as well. This looks similar to note lock which is funny as the problem originates from fixing it. Since we can pass hit timing to not get stuck on notes we haven't hit, we can miss notes we aren't supposed to hit yet. While it's fixable if you can react fast enough, it's not ideal.

The Theoretical Fix?
If we make circles have a secondary, barely larger automiss circle immediately under it that when clicked instead caused forcemiss() on the original circle, it would prevent extremely close misses from causing more damage than intended while drastic misses are kind of SOL.

Looking at some graphs, I think the automiss circle should have around 15% radius increase compared to the original circle but of course, this number is theoretical and not tested.

Side Effects?
- If you start streaming slightly earlier than you should while being "on time" with your aim you'll automiss a few notes
- If you're on time with your tapping but your aim is lagging streams/bursts will more likely automiss before accurately streaming the rest as opposed to getting a slew of yellows
- Streams/repeated bursts that skip notes can't be spammed through without potentially automissing

Solution for Side Effects?
Make it so the Automiss circle only sets a boolean to true, causing a click onto an unrelated circle will cause you to forcemiss() the last circle when true
 - should prevent you from automissing if you're slightly early with 

Remaining Side Effects?
- Aiming slightly too far ahead during streaming may cause a line of misses instead of skipping ahead to the next note

Thank you for your time and consideration.

**Comment by peppy** [comment]
I think you could hit accurately rather than adjusting the game to account for your weaknesses.

**Reply by ghost** [reply]
Actually no, I'm not just going to close the discussion and back off.

This post had nothing to do with getting the game to aim for me, I missed that first note. I KNOW I missed it. The problem is that the punishment for missing the first note escalated to the second note.

Nowhere in this post have I requested anything be made easier, just that the punishment doesn't overreach its boundaries. If anything the changes proposed by this post would make the game harder and more prone to a few misses before accurate streaming as opposed to handing out blanket yellows to keep someone's combo.

Difficulty is a combination of challenge and punishment, but when the punishment extends to an unrelated challenge the game feels unjust. That is why I posted in hope that I could get someone's attention to improve the game.

Even if my implementation here is horrible or the entire concept only leads to other problems in other areas, I'd prefer to have my idea thrown out in that context than as an insult to my mentality.

**Reply by ghost** [reply]
The statement that "[I] could just hit accurately rather than adjusting the game to account for [my] weaknesses." could also be extended to notelock, don't miss a note and there won't be a problem. Just git gud scrub.

In that context why even fix it? It's identical to this issue just on a grander scale, the punishment extended well beyond the capacity of what missing a single note should be, so it was fixed to be more reasonable.

While my problem won't force a full wipe, it's still an imperfection to the game that might be nice to fix if you have the time.

**Reply by ghost** [reply]
Thank you for your time and consideration.

**Comment by ghost** [comment]
Edited the post:
- Added the "Solution for Side Effects" and "Remaining Side Effects" portion
- Rephrasing the post from "near hit protection" to "auto miss for near hit" 
- Replaced the incessant use of forcemiss() with automiss for clarity.

**Reply by ghost** [reply]
Currently thinking of ideas to solve the problem with aiming barely too far with streaming

**Comment by Rekunan** [comment]
I feel like this is too niche of an issue to warrant the side effect(s) listed.
